crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -337,6 +337,66 @@ impl CelestiaClient {
         Ok(sequencer_namespace_datas)
     }
 
+    /// Returns the rollup data for a given rollup namespace at a given height, if it exists.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// + the verification key could not be constructed from the data stored in `namespace_data`;
+    /// + the RPC to fetch the blobs failed.
+    pub async fn get_rollup_data(
+        &self,
+        height: u64,
+        namespace_data: &SignedNamespaceData<SequencerNamespaceData>,
+        rollup_namespace: Namespace,
+    ) -> eyre::Result<Option<RollupNamespaceData>> {
+        let verification_key = VerificationKey::try_from(&*namespace_data.public_key)
+            .wrap_err("failed constructing verification key from stored bytes")?;
+
+        let req = GetAllRequest {
+            height,
+            namespace_ids: vec![*rollup_namespace],
+        };
+
+        let rsp = match self.client.blob_get_all(req).await {
+            Ok(rsp) => rsp,
+            Err(e) => {
+                if let ErrorKind::Rpc(astria_celestia_jsonrpc_client::JsonRpseeError::Call(inner)) =
```

Comment:
is there a clean way to shorten this line?

Reply:
@noot can you use the utility landed with #236 to make this a bit neater?